File: 002_palindromes/palind.py
#!/usr/bin/python

import sys

words = []

for line in sys.stdin:

  word = line.rstrip().lower()
  half = len(word) / 2

  is_palindrom = 1
  for num in range(1, half):

    first = num - 1
    last = -1 * num  

    if word[first] != word[last]:
      is_palindrom = 0
      break
  
  if is_palindrom: words.append(word)

for word in sorted(words):
  print word

File: 002_palindromes/palind.rb
#!/usr/bin/env ruby

words = []
ARGF.each do |line|
    word = line.chomp.downcase
    half = word.length / 2

    is_palindrom = 1
    for i in 1 .. half
        first = i - 1
        last  = i * -1

        if word[first] != word[last, 1]
            is_palindrom = 0
            break
        end
    end

    if is_palindrom > 0
        words.push(word)
    end
end

words.sort.each do |word|
    puts word
end
